compiler/qsc_parse/src/expr.rs

```diff
@@ -710,9 +711,20 @@ fn lambda_op(s: &mut ParserContext, input: Expr, kind: CallableKind) -> Result<B
     Ok(Box::new(ExprKind::Lambda(kind, input, output)))
 }
 
-fn field_op(s: &mut ParserContext, lhs: Box<Expr>) -> Result<Box<ExprKind>> {
+#[allow(clippy::unnecessary_wraps)]
+fn recovering_field_op(s: &mut ParserContext, lhs: Box<Expr>) -> Result<Box<ExprKind>> {
     s.expect(WordKinds::Field);
-    Ok(Box::new(ExprKind::Field(lhs, ident(s)?)))
+    let expr = ExprKind::Field(
+        lhs,
+        match ident(s) {
+            Ok(i) => FieldAccess::Ok(i),
+            Err(e) => {
+                s.push_error(e);
+                FieldAccess::Err
+            }
+        },
+    );
+    Ok(Box::new(expr))
```

Comment:
did we ever figure out why all of our parsers are boxed? or are you just matching conventions?

Reply:
Mostly matching convention.  @swernli and I had a conversation about this a while back. I'll try to give a summary:

- Early on, during some perf optimization work, we discovered that keeping struct sizes small, by boxing, saved a lot of time spent in stack allocation and moving things around.
- The strategy is mentioned in : https://nnethercote.github.io/perf-book/heap-allocations.html#box
- However, today, we hardly (if ever) move the AST around. Most of the work during compilation, as you probably already know, is done using a reference to _the_ AST.
- So we think these early perf decisions may not be relevant anymore, but we have not validated (or invalidated) them with current code. 

For my changes, in my attempt to please everyone, I used the rule of thumb "no structs in the AST larger than 24 bytes", so I boxed things as I hit that limit. The "24" number is purely based on vibes and not hard numbers.